lib/helpers/populate/assignVals.js

```diff
@@ -166,7 +165,7 @@ module.exports = function assignVals(o) {
     // If lean, need to check that each individual virtual respects
     // `justOne`, because you may have a populated virtual with `justOne`
     // underneath an array. See gh-6867
-    mpath.set(o.path, valueToSet, docs[i], lookupLocalFields, setValue, false);
+    return mpath.set(o.path, valueToSet, docs[i], lookupLocalFields, setValue, false);
```

Comment:
This causes every test that populates multiple docs to fail, because this is an early `return` that prevents every doc after the first from being populated. This change may accidentally fix this one use case, but it breaks a lot of others.